EPR/Library/noise.cpp

```diff
@@ -0,0 +1,264 @@
+const int P3 = 337;
+const double phi = 0.6180339;
+
+/********************** Definition for Noise *******************/
+Noise::Noise() {
+    int i;
+
+    for (i = 0; i < NUMPTS; ++i)
+        pts[i] = ((float)rand() / (float)RAND_MAX);
+}
+
+double Noise::noise(const Vector& p) {
+    int xi, yi, zi;
+    int xa, xb, xc, ya, yb, yc, za, zb, zc;
+    double xf, yf, zf;
+    double x2, x1, x0, y2, y1, y0, z2, z1, z0;
+    double p000, p100, p200, p010, p110, p210, p020, p120, p220;
+    double p001, p101, p201, p011, p111, p211, p021, p121, p221;
+    double p002, p102, p202, p012, p112, p212, p022, p122, p222;
+
+    xi = (int)floor(p.x);
+    xa = (int)floor(P1 * (xi * phi - floor(xi * phi)));
+    xb = (int)floor(P1 * ((xi + 1) * phi - floor((xi + 1) * phi)));
+    xc = (int)floor(P1 * ((xi + 2) * phi - floor((xi + 2) * phi)));
+
+    yi = (int)floor(p.y);
+    ya = (int)floor(P2 * (yi * phi - floor(yi * phi)));
+    yb = (int)floor(P2 * ((yi + 1) * phi - floor((yi + 1) * phi)));
+    yc = (int)floor(P2 * ((yi + 2) * phi - floor((yi + 2) * phi)));
+
+    zi = (int)floor(p.z);
+    za = (int)floor(P3 * (zi * phi - floor(zi * phi)));
+    zb = (int)floor(P3 * ((zi + 1) * phi - floor((zi + 1) * phi)));
+    zc = (int)floor(P3 * ((zi + 2) * phi - floor((zi + 2) * phi)));
+
+    p000 = pts[(xa + ya + za) & (NUMPTS - 1)];
+    p100 = pts[(xb + ya + za) & (NUMPTS - 1)];
+    p200 = pts[(xc + ya + za) & (NUMPTS - 1)];
+    p010 = pts[(xa + yb + za) & (NUMPTS - 1)];
+    p110 = pts[(xb + yb + za) & (NUMPTS - 1)];
+    p210 = pts[(xc + yb + za) & (NUMPTS - 1)];
+    p020 = pts[(xa + yc + za) & (NUMPTS - 1)];
+    p120 = pts[(xb + yc + za) & (NUMPTS - 1)];
+    p220 = pts[(xc + yc + za) & (NUMPTS - 1)];
+    p001 = pts[(xa + ya + zb) & (NUMPTS - 1)];
+    p101 = pts[(xb + ya + zb) & (NUMPTS - 1)];
+    p201 = pts[(xc + ya + zb) & (NUMPTS - 1)];
+    p011 = pts[(xa + yb + zb) & (NUMPTS - 1)];
+    p111 = pts[(xb + yb + zb) & (NUMPTS - 1)];
+    p211 = pts[(xc + yb + zb) & (NUMPTS - 1)];
+    p021 = pts[(xa + yc + zb) & (NUMPTS - 1)];
+    p121 = pts[(xb + yc + zb) & (NUMPTS - 1)];
+    p221 = pts[(xc + yc + zb) & (NUMPTS - 1)];
+    p002 = pts[(xa + ya + zc) & (NUMPTS - 1)];
+    p102 = pts[(xb + ya + zc) & (NUMPTS - 1)];
+    p202 = pts[(xc + ya + zc) & (NUMPTS - 1)];
+    p012 = pts[(xa + yb + zc) & (NUMPTS - 1)];
+    p112 = pts[(xb + yb + zc) & (NUMPTS - 1)];
+    p212 = pts[(xc + yb + zc) & (NUMPTS - 1)];
+    p022 = pts[(xa + yc + zc) & (NUMPTS - 1)];
+    p122 = pts[(xb + yc + zc) & (NUMPTS - 1)];
+    p222 = pts[(xc + yc + zc) & (NUMPTS - 1)];
+
+    xf = p.x - xi;
+    x1 = xf * xf;
+    x2 = 0.5 * x1;
+    x1 = 0.5 + xf - x1;
+    x0 = 0.5 - xf + x2;
+
+    yf = p.y - yi;
+    y1 = yf * yf;
+    y2 = 0.5 * y1;
+    y1 = 0.5 + yf - y1;
+    y0 = 0.5 - yf + y2;
+
+    zf = p.z - zi;
+    z1 = zf * zf;
+    z2 = 0.5 * z1;
+    z1 = 0.5 + zf - z1;
+    z0 = 0.5 - zf + z2;
+
+    return   z0 * (y0 * (x0 * p000 + x1 * p100 + x2 * p200) +
```

Comment:
It would be nice to have consistent indentation here

Reply:
#28 created

Reply:
fixed in commit 52321b2